test/ILogger.NetStandard.Tests/ILoggerIntegrationTests.cs

```diff
@@ -0,0 +1,296 @@
+                (telemetryItem, telemetryProcessor) => itemsReceived.Add(telemetryItem),
+                configureTelemetryConfiguration: null,
+                configureApplicationInsightsOptions: (appInsightsLoggerOptions) => appInsightsLoggerOptions.IncludeScopes = true);
+
+            ILogger<ILoggerIntegrationTests> testLogger = serviceProvider.GetRequiredService<ILogger<ILoggerIntegrationTests>>();
+
+            using (testLogger.BeginScope("TestScope"))
+            {
+                using (testLogger.BeginScope<IReadOnlyCollection<KeyValuePair<string, object>>>(new Dictionary<string, object> { { "Key", "Value" } }))
+                {
+                    testLogger.LogInformation("Testing");
+                    testLogger.LogError(new Exception("TestException"), "Exception");
+                }
+            }
+
+            Assert.AreEqual(" => TestScope", (itemsReceived[0] as ISupportProperties).Properties["Scope"]);
+            Assert.AreEqual("Value", (itemsReceived[0] as ISupportProperties).Properties["Key"]);
+
+            Assert.AreEqual(" => TestScope", (itemsReceived[1] as ISupportProperties).Properties["Scope"]);
+            Assert.AreEqual("Value", (itemsReceived[1] as ISupportProperties).Properties["Key"]);
+        }
+
+        /// <summary>
+        /// Ensures that the <see cref="ApplicationInsightsLoggerOptions.IncludeScopes"/> switch is honored and scopes are excluded
+        /// when switch is false.
+        /// </summary>
+        [TestMethod]
+        [TestCategory("ILogger")]
+        public void ApplicationInsightsLoggerAddsScopeWhenSwitchIsFalse()
+        {
+            List<ITelemetry> itemsReceived = new List<ITelemetry>();
+
+            // Case where Scope is NOT Included
+            IServiceProvider serviceProvider = ILoggerIntegrationTests.SetupApplicationInsightsLoggerIntegration(
+                (telemetryItem, telemetryProcessor) => itemsReceived.Add(telemetryItem),
+                configureTelemetryConfiguration: null,
+                configureApplicationInsightsOptions: (appInsightsLoggerOptions) => appInsightsLoggerOptions.IncludeScopes = false);
+
+            ILogger<ILoggerIntegrationTests> testLogger = serviceProvider.GetRequiredService<ILogger<ILoggerIntegrationTests>>();
+
+            using (testLogger.BeginScope("TestScope"))
+            {
+                using (testLogger.BeginScope<IReadOnlyCollection<KeyValuePair<string, object>>>(new Dictionary<string, object> { { "Key", "Value" } }))
+                {
+                    testLogger.LogInformation("Testing");
+                    testLogger.LogError(new Exception("TestException"), "Exception");
+                }
+            }
+
+            Assert.IsFalse((itemsReceived[0] as ISupportProperties).Properties.ContainsKey("Scope"));
+            Assert.IsFalse((itemsReceived[0] as ISupportProperties).Properties.ContainsKey("Key"));
+
```

Comment:
lets check the actual message in the itemReceived as well. and the count is also as expected.